core/services/ocr2/plugins/ocr2keeper/evm21/registry_check_pipeline.go

```diff
@@ -103,6 +103,11 @@ func (r *EvmRegistry) verifyCheckBlock(_ context.Context, checkBlock, upkeepId *
 	var ok bool
 	// verify check block number and hash are valid
 	h, ok = r.bs.queryBlocksMap(checkBlock.Int64())
+	// if this block number/hash combo exists in block subscriber, this check block and hash still exist on chain and are valid
+	if ok && h == checkHash.Hex() {
+		r.lggr.Debugf("check block hash %s exists on chain at block number %d for upkeepId %s", checkHash.Hex(), checkBlock, upkeepId)
+		return encoding.NoPipelineError, false
+	}
 	if !ok {
```

Comment:
we'll need to remove this if condition, we want to do also check in the case hash existed but didn't match